tools.go

```diff
@@ -0,0 +1,52 @@
+package resolver
+
+import (
+	"net"
+
+	"github.com/miekg/dns"
+)
+
+func validateResp(server, name string, r *dns.Msg, err error) error {
+	if err != nil {
+		// TODO: extract information from err.(type)
+		var timeout bool
+		var temporary bool
+		var notfound bool
+
+		return &net.DNSError{
+			Err:         err.Error(),
+			Server:      server,
+			Name:        name,
```

Comment:
if you pass a `*dns.Msg` why you need a `name`? What is wrong with `*dns.Msg.Question[0].Name`?

Reply:
Nothing wrong, just ignorance

Reply:
btw, don't be mislead by the `0` in `Question[0]`. Even if the standard permits multiple `Question` bundle, afaik there is no DNS sever implementation that supports that, so `Question[0]` is pretty much the only `Question`

Reply:
Amended